admin-dev/themes/new-theme/js/components/form/form-object-mapper.js

```diff
@@ -0,0 +1,173 @@
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+const {$} = window;
+
+/**
+ * This is able to watch an HTML form and parse it as a Javascript object based on a configurable
+ * mapping. Each field from the model is mapped to a form input, or several, each input is watched
+ * to keep the model consistent.
+ *
+ * When a model field has several inputs associated it keeps the in sync when either of them is updated.
+ */
+export default class FormObjectMapper {
+  /**
+   * @param $form {jQuery}
+   * @param modelMapping {Object}
+   * @param eventEmitter {EventEmitter}
+   * @param config {Object}
+   */
+  constructor($form, modelMapping, eventEmitter, config) {
+    this.$form = $form;
+    this.fullModelMapping = modelMapping;
+    this.eventEmitter = eventEmitter;
+
+    const inputConfig = config || {};
+
+    // This event is registered so when it is triggered it forces the object update
+    this.updateModelEventName = inputConfig.updateModel || 'updateModel';
+    // This event is emitted each time the object is updated (from both input change and external event)
+    this.modelUpdatedEventName = inputConfig.modelUpdated || 'modelUpdated';
+
+    this.modelMapping = {};
+    this.formMapping = {};
+    this.object = {};
+
+    this.initFormMapping();
+    this.watchUpdates();
+    this.updateFullObject();
+  }
+
+  getObject() {
+    return this.object;
+  }
+
+  /**
+   * Watches if changes happens from the form or via an event.
+   */
+  watchUpdates() {
+    this.$form.on('keyup change dp.change', ':input', (event) => this.inputUpdated(event));
+    this.eventEmitter.on(this.updateModelEventName, () => this.updateFullObject());
+  }
+
+  /**
+   * Triggered when a form input has been changed.
+   *
+   * @param event {Object}
+   */
+  inputUpdated(event) {
+    const target = event.currentTarget;
+
+    if (!Object.prototype.hasOwnProperty.call(this.formMapping, target.name)) {
+      return;
+    }
+
+    const updatedValue = $(target).val();
+    const updatedModelKey = this.formMapping[target.name];
+    const modelInputs = this.fullModelMapping[updatedModelKey];
+
+    // Update linked inputs
+    modelInputs.forEach((inputName) => {
+      if (inputName === target.name) {
+        return;
```

Comment:
```suggestion
        return;
```

Returning inside a forEach looks really weird to me, but it works (closing the callback execution), why not